src/Symfony/Component/Yaml/Inline.php

```diff
@@ -712,8 +717,12 @@ private static function evaluateScalar($scalar, $flags, $references = [])
                         return null;
                     case 0 === strpos($scalar, '!php/const'):
                         if (self::$constantSupport) {
+                            if ($isMappingKey) {
```

Comment:
One edge case is if the constant '' is defined (define('', 'foo')). In this case it currently works but the resulting key value is the tag (!php/const), not the evaluated value. So I guess we don't care of breaking this case.